VitDeck/Assets/VitDeck/Validator/Rules/Booth/B04_ErrorShaderRule.cs

```diff
@@ -0,0 +1,49 @@
+﻿using UnityEngine;
+using System.Linq;
+
+namespace VitDeck.Validator
+{
+    /// <summary>
+    /// エラーシェーダーの検出ルール
+    /// </summary>
+    /// <remarks>シェーダーエラーの存在するオブジェクトを検出する。</remarks>
+    public class ErrorShaderRule : BaseRule
+    {
+        /// <summary>
+        /// コンストラクタ。
+        /// </summary>
+        /// <param name="name">ルール名</param>
+        public ErrorShaderRule(string name) : base(name) { }
+
+        protected override void Logic(ValidationTarget target)
+        {
+            foreach (var obj in target.GetAllObjects())
+            {
+                if (obj.GetComponent<Renderer>() == null)
+                    continue;
+
+                var materials = obj.GetComponent<Renderer>().sharedMaterials;
+
+                foreach (var material in materials)
+                {
+                    if (material == null)
+                    {
+                        AddIssue(new Issue(obj, IssueLevel.Error, string.Format("オブジェクト({0})のRendererにマテリアルの参照がありません。", obj.name), string.Empty, string.Empty));
```

Comment:
1. マテリアルの参照なしをここでErrorとして検出してしまうと
ParticleRendererのようなすべてにマテリアルが設定されていない可能性のあるRendererで困る場合がありそうなのでレベルをError→Infoにするのはどうでしょうか？

![image](https://user-images.githubusercontent.com/46106857/59976498-5e6ed900-9600-11e9-8d78-62e9dd90a991.png)

2. また下コメントのスクショのようにメッセージ内でオブジェクト名が重複するので削った方が読みやすそうです。
```suggestion
                        AddIssue(new Issue(obj, IssueLevel.Info, "オブジェクトのRendererにマテリアルの参照がありません。", string.Empty, string.Empty));
```